tracking/translations_parser/cli/taskcluster.py

```diff
@@ -113,40 +163,60 @@ def main() -> None:
         with args.input_file.open("r") as f:
             lines = (line.strip() for line in f.readlines())
 
+    # Load secret from Taskcluster and auto-configure naming
+    if args.taskcluster_secret:
+        assert os.environ.get(
+            "TASKCLUSTER_PROXY_URL"
+        ), "When using `--taskcluster-secret`, `TASKCLUSTER_PROXY_URL` environment variable must be set too."
+
+        # Weight and Biases client use environment variable to read the token
+        os.environ.setdefault("WANDB_API_KEY", get_wandb_token(args.taskcluster_secret))
+
+        project_name, group_name, run_name = get_wandb_names()
+    else:
+        # Fallback to CLI args for names
+        project_name = args.wandb_project
+        group_name = args.wandb_group
+        run_name = args.wandb_run_name
+
+    # Enable publication on weight and biases when project is set
     publishers: list[Publisher] = [CSVExport(output_dir=args.output_dir)]
-    if args.wandb_project:
+    if project_name:
         publishers.append(
             WandB(
-                project=args.wandb_project,
+                project=project_name,
+                group=group_name,
+                name=run_name,
                 artifacts=args.wandb_artifacts,
-                group=args.wandb_group,
                 tags=["cli"],
```

Comment:
I see that the tag "cli" is visible in the workgroup. I see we also use the tag "Taskcluster" in the code. Should we utilize tags to distinguish between different modes of publishing? We have 3 now: GCP, taskcluster-offline and taskcluster online.

Reply:
I exposed the tags through a new CLI argument `--tags`:
- on CI uploads, it will use `taskcluster`
- on offline group upload using the dedicated entry point, it will use `taskcluster-offline`
- when uploading a local folder, the operator should set `GCP`